ion/marshal.go

```diff
@@ -446,8 +445,9 @@ func (m *Encoder) encodeTimeDate(v reflect.Value) error {
 	// Get number of fractional seconds precisions
 	ns := t.Nanosecond()
 	numFractionalSeconds := 0
+	// Go native time object will always have 9 precision units when nanoseconds > 0
```

Comment:
Why does the precision change when `nanoseconds > 0`? Why isn't it a 0 of the same precision?

Reply:
For `time.time` objects, if the nanosecond is 0 or not specified, it is treated as no nanoseconds.

`timeObject, err := time.Parse(time.RFC3339, "2000-01-01T01:01:01.00Z")` will have timeObject.nanoseconds = 0

`timeObject, err := time.Parse(time.RFC3339, "2000-01-01T01:01:01.01Z")` will have timeObject.nanoseconds = 10000000

Reply:
The [`Time` type documentation](https://golang.org/pkg/time/#Time) says:

> A Time represents an instant in time with nanosecond precision.

Unlike Ion's `Timestamp` type, which supports variable precision, it seems like `time.Time` _always_ has nanosecond precision. If that's true, I would expect a `time.Time` with zero nanoseconds to be written out as a `Timestamp` with a fractional seconds precision of 9 decimal places.

> For time.time objects, if the nanosecond is 0 or not specified, it is treated as no nanoseconds.

Could you please point me to the relevant docs for this? How do we distinguish between `0` or `not specified` in a `time.Time`?

Reply:
You are correct, I was basing my interpretation on the fact that time.nanoseconds = 0 and the string representation was not showing fractional seconds 🤦 .

I removed the `nanoseconds > 0` check in  `encodeTimeDate()`.

I also added guards to  `NewDateTimestamp()`, `NewTimestamp()` and set `fractionalPrecision = 9` if the timestamp precision is `TimestampPrecisionNanosecond`